packages/markdown-transform/lib/transform.js

```diff
@@ -165,6 +166,10 @@ const transformationGraph = {
         pdfmake: (input, parameters, options) => {
             return PdfTransformer.ciceroMarkToPdfMake(input, options);
         },
+        ooxml: (input) => {
+            const t = new OOXMLTransformer();
+            return t.toOOXML(input).ooxml;
```

Comment:
Do we need the `ooxml` child attribute here? The other transformations just return the output.

Reply:
@dselman 
Yes, actually `OOXMLTransfomer` returns `{ooxml, variableFrequency}`.

Reply:
What is variable frequency and why does the caller need it?

Reply:
It returns the count of different variables present in a template. 
We need it because there is a requirement to provide a unique identifier to `rich text input` in MS-WORD which is generated using this.

Reply:
> What is variable frequency and why does the caller need it?

MS Word would require it for rendering all the variables.

Reply:
Can you explain why, please? How does variable frequency help?

Reply:
Sure!

In MS Word, `org.accordproject.ciceromark.Variable` is converted to a `w:sdt` XML tag. This XML tag represents content controls in MS Word. A content control has many attributes but we use only two of them - `title` and `tag` (think of them as analogous to `id` and `class` in HTML, respectively).

The `title` attribute is compulsory and **has to be unique** but since there can be many instances of a variable in a clause, we cannot assign each of them the same title. For example, all four instances of `Shipper` in `acceptance-of-delivery` cannot have the same title `Shipper` so I decided to maintain a counter object which would keep a count of the occurrences of the variable and assign the title as - `Shipper1`, `Shipper2`, `Shipper3`, and `Shipper4`. We are also exporting the `counter` variable as MS Word will use it to attach listeners [here](https://github.com/accordproject/cicero-word-add-in/blob/master/src/components/TemplateLibrary/index.js#L146).

In retrospect, I think exporting counter variable _may not_ be required. I will try to look into it and discuss this with you and @K-Kumar-01 in the next meeting.

@K-Kumar-01 , for now, maybe we can revert exporting of `counter` object and do it when we really find that necessary.

Reply:
@algomaster99 I dont like the idea of not exporting the `counter`. Reason being i do believe that we would need it a later point, basically when we use it with add-in. On what basis so you think that we might not require the variable? Could you please explain that?

Reply:
Cicero Word add-in uses the counter variable to attach listeners. Instead of the for-loop, we can fetch all the content controls using [`body.contentControls`](https://docs.microsoft.com/en-us/javascript/api/word/word.body?view=word-js-preview#contentControls) and then attach the listener one-by-one. However, there are two questions to ponder over first:
1. What happens if the event listener is attached again to a content control which already had an event listener attached?
2. What happens when the content control with a non-unique title is inserted?

Reply:
It should be possible to generate a unique id for a variable based on its position in the tree. E..g using the path to the node. Alternatively we might be able to assign a UUID further up in the transformation stack. Let's discuss some ideas.

Reply:
Generating a unique idea is not the problem but getting back to attach event listeners is. Moreover, we want to keep this ID as close as possible to the actual variable name as they are means to inform the client which variable they are interacting with.

<img width="87" alt="cc" src="https://user-images.githubusercontent.com/35191225/126362819-db308cdd-dc1b-4f47-b094-d8a40aff06b4.PNG">

`Shipper1` is the title and if someone edits, it would mean that they want to modify the shipper variable.

Reply:
I am looking for ways to make us _not require_ to export the `counter` variable. But we shall use it for OOXML generation as we want to make the content control unique.

Reply:
@K-Kumar-01 I think you can remove exporting of `counter`. I can demonstrate how we can do without it in the next meeting.

Reply:
@algomaster99 
That would need refactoring first then. I will create a separate PR that will deal with the above.
Once that is done, I will merge it.
Since, currently only windows are available on my pc it might take some time to close these PRs.

PS: HDD corruption made Linux vanish 😢. Will reinstall it.

Reply:
Sure. No problem.